File: python/binary_search.py
list = list(range(1, 101))


def binary_search(ordered_list, searched_value):
    low = 0
    high = len(ordered_list) - 1
    step = 0
    while low <= high:
        middle = int((low + high) / 2)

        if ordered_list[middle] == searched_value:
            step += 1
            print('This is BS step count: ' + str(step))
            return 'BS result: ' + str(ordered_list[middle])
        elif ordered_list[middle] > searched_value:
            step += 1
            high = middle - 1
        else:
            step += 1
            low = middle + 1
    print('This is BS step count: ' + str(step))
    return None


def simple_search(ordered_list, searched_value):
    step = 0
    for num in ordered_list:
        if num == searched_value:
            step += 1
            print('This is SS step count: ' + str(step))
            return 'SS result: ' + str(num)
        else:
            step += 1
    print('This is SS step count: ' + str(step))
    return None


result_ss = simple_search(list, 87)
result_bs = binary_search(list, 87)
print(result_ss)
print(result_bs)

File: javascript/binarySearch.js
// This creates an array filled with consecutive numbers up to 20
let n = 21;
const list = Array.from(Array(n).keys());

function binarySearch(orderedArray, searchedValue) {
    let low = 0;
    let high = orderedArray.length - 1;

    while(low <= high) {
        let middle = Math.floor((low + high) / 2);

        if (orderedArray[middle] === searchedValue) {
            return orderedArray[middle];
        }
        if(orderedArray[middle] > searchedValue) {
            high = middle - 1;
        } else {
            low = middle + 1;
        }
    }
    return 'Not in Array';
}

let value = binarySearch(list, 0);
console.log(value);